error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/model-1.rs:28:11
   |
28 | #[ensures(!(x.model().a == x.model().b))]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/model-1.rs:29:1
   |
29 | fn test1(x: X) {}
   | ^^^^^^^^^^^^^^^^^
note: counterexample for "x"
       value:   X_model {
            a: 5,
            b: 5,
        }
  --> $DIR/model-1.rs:29:10
   |
29 | fn test1(x: X) {}
   |          ^
note: counterexample for "result"
       value:   ()
  --> $DIR/model-1.rs:29:16
   |
29 | fn test1(x: X) {}
   |                ^^

error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/model-1.rs:32:11
   |
32 | #[ensures(!(x.model().b == y.model().a))]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/model-1.rs:33:1
   |
33 | fn test2(x: X, y: X) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^
note: counterexample for "x"
       value:   X_model {
            a: 2,
            b: 5,
        }
  --> $DIR/model-1.rs:33:10
   |
33 | fn test2(x: X, y: X) {}
   |          ^
note: counterexample for "y"
       value:   X_model {
            a: 5,
            b: 2,
        }
  --> $DIR/model-1.rs:33:16
   |
33 | fn test2(x: X, y: X) {}
   |                ^
note: counterexample for "result"
       value:   ()
  --> $DIR/model-1.rs:33:22
   |
33 | fn test2(x: X, y: X) {}
   |                      ^^

error: aborting due to 2 previous errors

